app/kuma-dp/pkg/dataplane/metrics/profiles.go

```diff
@@ -0,0 +1,236 @@
+		Type:  v1alpha1.ExactSelectorType,
+		Match: "envoy_cluster_membership_degraded",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ExactSelectorType,
+		Match: "envoy_cluster_membership_healthy",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ContainsSelectorType,
+		Match: "error",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ContainsSelectorType,
+		Match: "fail",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ContainsSelectorType,
+		Match: "reset",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ContainsSelectorType,
+		Match: "eject",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ContainsSelectorType,
+		Match: "overflow",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ContainsSelectorType,
+		Match: "cancelled",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ContainsSelectorType,
+		Match: "max_duration_reached",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ContainsSelectorType,
+		Match: "no_cluster",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ContainsSelectorType,
+		Match: "no_route",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ContainsSelectorType,
+		Match: "no_filter_chain",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ContainsSelectorType,
+		Match: "reject",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ContainsSelectorType,
+		Match: "denied",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ContainsSelectorType,
+		Match: "days_until_first_cert_expiring",
+	}),
+	// end of golden signals
+	// start of dashboards
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.PrefixSelectorType,
+		Match: "envoy_server",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.PrefixSelectorType,
+		Match: "envoy_control_plane",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ContainsSelectorType,
+		Match: "rbac",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ContainsSelectorType,
+		Match: "cx_active",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ContainsSelectorType,
+		Match: "cx_connect",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ExactSelectorType,
+		Match: "envoy_cluster_ssl_handshake",
+	}),
+	selectorToFilterFunction(v1alpha1.Selector{
+		Type:  v1alpha1.ExactSelectorType,
+		Match: "envoy_cluster_membership_total",
+	}),
+	// end of dashboards
+}
+
+func ProfileMutatorGenerator(sidecar *v1alpha1.Sidecar) PrometheusMutator {
+	effectiveSelectors := []selectorFunction{alwaysSelect} // default is All
+	if sidecar != nil && sidecar.Profiles != nil && sidecar.Profiles.AppendProfiles != nil && len(*sidecar.Profiles.AppendProfiles) == 1 {
```

Comment:
if I understand it correctly we will only use 1st profile? Don't we want to support setting multiple profiless?

Reply:
If I remember correctly since we only have 3 profiles now we said that we will only allow 1 profile and allow multiple profiles in the future, that's why this validation is here https://github.com/kumahq/kuma/pull/9579/files#diff-e28548cca6e7c15cb7915892c16c7b8a4916a29ab0004f64c7a4f3c1da18b4c5R44

Reply:
right, with current set of profiles we don't need this